services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParser.java

```diff
@@ -294,26 +319,6 @@ public VenicePath parseResourceUri(String uri, HTTP_REQUEST request) throws Rout
         }
       }
 
-      boolean decompressOnClient = routerConfig.isDecompressOnClient();
```

Comment:
How do we dodge needing this?  Is it because this decision is delegated to the responseDecompressor?

Reply:
This code is moved to a separate function. See line 401.